cabal.project

```diff
@@ -54,282 +66,10 @@ package cryptonite
   -- generation is dubious. Set the flag so we use /dev/urandom by default.
   flags: -support_rdrand
 
--- ---------------------------------------------------------
--- Disable all tests by default
-
-tests: False
+tests: True
```

Comment:
Do we need this? The [documentation](https://cabal.readthedocs.io/en/stable/cabal-project.html#cfg-field-tests) seems to suggest it is not useful:

> Force test suites to be enabled. For most users this should not be needed, as we always attempt to solve for test suite dependencies, even when this value is False; furthermore, test suites are automatically enabled if they are requested as a built target.

Reply:
Hmm, I think in the past it used to be necessary so cabal wouldn't re-solve for tests when you switched to running tests, perhaps they fixed that. But that's an orthogonal issue I guess.

Reply:
yes, before you had to do `tests: False` and enabled them one by one for the local packages. This is now solved (which is why I left `tests: True` which was the original intent). Now I wonder if one needs to enable them at all.

Reply:
When using `source-repository-package` the directive `tests: True` turns on tests within all `source-repository-package` dependencies, which (especially higher up the stack) just needless runs tests in dependencies for which it should be possible to avoid re-rumning tests. 

In say `db-sync` we have `tests: False` to disable all tests and the specific stanzas to turn on tests for packages in the `db-sync` repo.

Reply:
> When using source-repository-package the directive tests: True turns on tests within all source-repository-package dependencies, which (especially higher up the stack) just needless runs tests in dependencies for which it should be possible to avoid re-rumning tests.

Is this still true in recent versions of cabal? We have `tests: True` in `plutus` and I've never observed it running or building tests for s-r-ps, and I tried again just now.